platform/core.output2/src/org/netbeans/core/output2/options/OutputSettingsPanel.java

```diff
@@ -583,27 +578,21 @@ private void initPreview() {
         }
         previewInputOutput.getOut().close();
         previewInputOutput.getErr().close();
-        outputOptions.addPropertyChangeListener(new PropertyChangeListener() {
-            @Override
-            public void propertyChange(PropertyChangeEvent evt) {
-                if (!evt.getPropertyName().equals(OutputOptions.PROP_INITIALIZED)
-                        && outputOptions.isInitialized()) {
-                    controller.changed(outputOptions.isChanged());
-                }
-                updateControlsByModel();
+        outputOptions.addPropertyChangeListener((PropertyChangeEvent evt) -> {
+            if (!evt.getPropertyName().equals(OutputOptions.PROP_INITIALIZED)
+                    && outputOptions.isInitialized()) {
+                controller.changed(outputOptions.isChanged());
             }
+            updateControlsByModel();
         });
-        SwingUtilities.invokeLater(new Runnable() {
-            @Override
-            public void run() {
-                Component component = previewPanel.getComponent(0);
-                if (component instanceof AbstractOutputTab) {
-                    ((AbstractOutputTab) component).inputSent("Input from keyboard");
-                }
-                try {
-                    in.close();
-                } catch (IOException ex) {}
+        SwingUtilities.invokeLater(() -> {
+            Component component1 = previewPanel.getComponent(0);
+            if (component1 instanceof AbstractOutputTab) {
+                ((AbstractOutputTab) component1).inputSent("Input from keyboard");
             }
+            try {
+                in.close();
```

Comment:
Just a nuance, caught my attention to get a reader closed in AWT, made me worry if we do write on that stream on a different thread.

Removing the otherwise unused `in` variable and use `previewInputOutput.getIn().close()` would make this more clear.

Reply:
yeah this looks weird but I believe `getIn()` does more than just opening a stream:
```
If a reader is ever requested, an input line is added to the tab and used to read one line at a time.
```
this sounds like it makes a visual difference and is the reason why it is opened that early and closed once it displayed something - but i could be wrong.